src/core/ht.c

```diff
@@ -135,136 +148,147 @@ void get_imu_data(int bCalcDiff)
     }
 }
 
+static void timer_callback_imu(union sigval timer_data)
+{
+    get_imu_data(true);
+    calc_ht();
+}
 
 /////////////////////////////////////////////////////////////////////////////////
 //HT function
 void init_ht()
 {
     ht_data.tiltAngle = 0;      
-    ht_data.tiltAngleLP = 0;    
-
     ht_data.rollAngle = 0;       
-    ht_data.rollAngleLP = 0;    
-
     ht_data.panAngle = 0;       
-    ht_data.panAngleLP = 0;     
 
-    ht_data.tiltRollBeta = 0.75;
-    ht_data.panBeta = 0.75; 
-    
-    ht_data.tiltInverse = -1; 
+    ht_data.tiltInverse = 1; 
     ht_data.rollInverse = -1; 
     ht_data.panInverse = -1; 
 
-    ht_data.tiltFactor = 17;
-    ht_data.rollFactor = 17;
-    ht_data.panFactor = 17;
-    
-    ht_data.tiltMaxPulse = 500;
-    ht_data.tiltMinPulse = -500; 
-    ht_data.tiltCenter = 1500; 
-    ht_data.panMaxPulse = 500;
-    ht_data.panMinPulse = -500; 
-    ht_data.panCenter = 1500; 
-    ht_data.rollMaxPulse = 500;
-    ht_data.rollMinPulse = -500; 
-    ht_data.rollCenter = 1500; 
-
     ht_data.htChannels[0] = 0;
     ht_data.htChannels[1] = 0;
     ht_data.htChannels[2] = 0;
     
-    ht_data.is_calibrated = 0;
     ht_data.enable = 0;
-}
-
-static void calc_gyr(float* gyrAngle) //in degree
-{
-    gyrAngle[0] = gyr_to_dps(ht_data.sensor_data.gyr.x - ht_data.gyr_offset[0]);
-    gyrAngle[1] = gyr_to_dps(ht_data.sensor_data.gyr.y - ht_data.gyr_offset[1]);
-    gyrAngle[2] = gyr_to_dps(ht_data.sensor_data.gyr.z - ht_data.gyr_offset[2]);
-}
-
-static void calc_acc(float* accAngle) //in G
-{
-    accAngle[0] = acc_to_g(ht_data.sensor_data.acc.x - ht_data.acc_offset[0]); 
-
-    accAngle[1] = acc_to_g(ht_data.sensor_data.acc.y - ht_data.acc_offset[1]);
+    set_maxangle_ht(g_setting.ht.max_angle);
+    ht_data.acc_offset[0] = g_setting.ht.acc_x;
+    ht_data.acc_offset[1] = g_setting.ht.acc_y;
+    ht_data.acc_offset[2] = g_setting.ht.acc_z;
+    ht_data.gyr_offset[0] = g_setting.ht.gyr_x;
+    ht_data.gyr_offset[1] = g_setting.ht.gyr_y;
+    ht_data.gyr_offset[2] = g_setting.ht.gyr_z;
+
+    // start timer
+    timer_t timerId = 0;
+    struct sigevent sev = { 0 };
+    struct itimerspec its = {   .it_value.tv_sec  = 1,
+                                .it_value.tv_nsec = 0,
+                                .it_interval.tv_sec  = 0,
+                                .it_interval.tv_nsec = 10000000
+                            };
+
+    sev.sigev_notify = SIGEV_THREAD;
+    sev.sigev_notify_function = &timer_callback_imu;
+
+    int res = timer_create(CLOCK_REALTIME, &sev, &timerId);
+    if (res != 0){
+        LOGE("Error timer_create: %s\n", strerror(errno));
+        return;
+    }
 
-    accAngle[2] = acc_to_g(ht_data.sensor_data.acc.z - ht_data.acc_offset[2]);
+    res = timer_settime(timerId, 0, &its, NULL);
```

Comment:
lovely. we should consider adopting this same pattern for the other interval based threads.